spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/SimpleConnection.java

```diff
@@ -71,15 +74,41 @@ public void close() {
 
 	@Override
 	public boolean isOpen() {
-		return this.delegate != null
-				&& (this.delegate.isOpen() || this.delegate.getClass().getSimpleName().contains("AutorecoveringConnection"));
+		if (this.delegate instanceof AutorecoveringConnection && !this.delegate.isOpen()) {
+			throw new AmqpException("Auto recovery connection is not currently open");
```

Comment:
My general concern here that we never `try...catch` this `isOpen()` and many our algorithm on that may end up unexpectedly. Like that `RabbitAdminIntegrationTests` on the `this.rabbitAdmin.declareExchange(exchange);`